tsdb/head.go

```diff
@@ -231,6 +235,26 @@ func newHeadMetrics(h *Head, r prometheus.Registerer) *headMetrics {
 	return m
 }
 
+const cardinalityCacheExperationTime int64 = 30 //seconds
+
+//Return top 10 Highest Cardinality stats for Labels and Value Names
+func (h *Head) PostingsCardinalityStats(statsByLabelName string) *index.PostingsStats {
+	h.cardinalityMutex.Lock()
+	defer h.cardinalityMutex.Unlock()
+
+	seconds := time.Now().Unix() - h.lastPostingsStatsCall
```

Comment:
It would be nice to have all of this logic using `time.Time` and `time.Duration` rather than various `int64`s.